linode/obj/resource.go

```diff
@@ -91,6 +91,16 @@ func readResource(ctx context.Context, d *schema.ResourceData, meta any) diag.Di
 		d.Set("endpoint", endpoint)
 	}
 
+	// Compute object access credentials from provider metadata when they're not configured explicitly
+	config := meta.(*helper.ProviderMeta).Config
+	if _, ok := d.GetOk("access_key"); !ok {
+		d.Set("access_key", config.ObjAccessKey)
+	}
+
+	if _, ok := d.GetOk("secret_key"); !ok {
+		d.Set("secret_key", config.ObjSecretKey)
+	}
+
```

Comment:
Can we just use the keys in S3 API calls without saving it? When a user config a provider level key, if the key is saved into the state of this resource, later when the user revoke the key and regenerate a new key, then they won't be able set it to provider again because this resource will continue used the saved old key.

Another concern is that by saving the provider level key into the state, the user won't be able to easily remove the key. If they just remove the key from provider without inspecting the state, they might left sensitive key in the state but think they already cleaned up everything sensitive.

Reply:
It's a good point. I was hesitating whether to save the keys into the resource state. I eventually saved it because the `getObject()` in the resource_test needs to get the access and secret keys from the state, and it seems no way from there to the provider config. But now I think I probably can generate temporary keys to get around it.